### PR TITLE
Gateway Documentation Overhaul

### PR DESCRIPTION
**Pull request by arHSM**
Websockets Websockets Websockets Websockets

### CODE REVIEW

File: pages/reference.mdx [hunk 1]
```diff
@@ -251,6 +251,122 @@ The HTTP API implements a process for limiting and preventing excessive requests
 
 Certain endpoints in the API are documented to accept booleans for their query string parameters. While there is no standard system for boolean representation in query string parameters, Discord represents such cases using `True`, `true`, or `1` for true and `False`, `false` or `0` for false.
 
+### Client Properties
+
+Client properties contain information about the current client and device. Some endpoints require them to be passed as a base64 encoded `X-Super-Properties` header to function properly. They are also required to [Identify](/topics/gateway-events#identify).
```
Comment (splatterxl) [inline pages/reference.mdx:256]: This is vague, should we elaborate on the super props header?

File: pages/reference.mdx [hunk 2]
```diff
@@ -251,6 +251,122 @@ The HTTP API implements a process for limiting and preventing excessive requests
...
+| browser_user_agent ^1^    | string                                                          | The current User-Agent                                                                             |
+| browser_version ^1^       | string                                                          | The current browser version                                                                        |
+| window_manager?           | string                                                          | Linux only. Output of `lsb_release -ds`                                                            |
+| distro?                   | string                                                          | Linux only. `(env.XDG_CURRENT_DESKTOP \|\| "unknown") + "," + (env.GDMSESSION \|\| "unknown")`     |
+| referrer?                 | string                                                          |                                                                                                    |
+| referrer_current?         | string                                                          | The current HTTP Referer                                                                           |
```
Comment (splatterxl) [inline pages/reference.mdx:277]:
```suggestion
| referrer_current?         | string                                                          | The referring source, from the [`Referer` HTTP header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referer)                                                                       |
```

File: pages/reference.mdx [hunk 3]
```diff
@@ -251,6 +251,122 @@ The HTTP API implements a process for limiting and preventing excessive requests
...
+| referring_domain?         | string                                                          |                                                                                                    |
+| referring_domain_current? | string                                                          | The current HTTP Referer domain                                                                    |
```
Comment (splatterxl) [inline pages/reference.mdx:279]:
```suggestion
| referring_domain_current? | string                                                          | The referring domain, from the HTTP Referer                                                  |
```

File: pages/reference.mdx [hunk 4]
```diff
@@ -251,6 +251,122 @@ The HTTP API implements a process for limiting and preventing excessive requests
...
+| search_engine?            | [search engine](#search-engine)                                 | Search engine matched from the HTTP Referer                                                        |
+| mp_keyword?               | string                                                          | Value of the `q` or `p` (for yahoo) parameter for the search engine which lead to the discord page |
```
Comment (splatterxl) [inline pages/reference.mdx:281]:
```suggestion
| mp_keyword?               | string                                                          | The search engine keyword that lead to the Discord page, from the `q` or `p` (Yahoo) search query  |
```

File: pages/reference.mdx [hunk 5]
```diff
@@ -251,6 +251,122 @@ The HTTP API implements a process for limiting and preventing excessive requests
...
+
+| Value          | Description |
+| -------------- | ----------- |
+| Android        |             |
+| BlackBerry     |             |
+| Mac OS X       |             |
+| iOS            |             |
+| Linux          |             |
+| Windows Mobile |             |
+| Windows        |             |
```
Comment (splatterxl) [inline pages/reference.mdx:303]: Do we need this empty description column?

File: pages/reference.mdx [hunk 6]
```diff
@@ -251,6 +251,122 @@ The HTTP API implements a process for limiting and preventing excessive requests
...
+###### Release Channel
+
+{/* TODO: Figure out iOS stable & TestFlight */}
+
+| Value         | Description    |
+| ------------- | -------------- |
+| stable        | Desktop Stable |
+| ptb           | Desktop PTB    |
+| canary        | Desktop Canary |
+| staging       |                |
```
Comment (splatterxl) [inline pages/reference.mdx:338]:
```suggestion
| staging       | Discord Staging  |
```

File: pages/reference.mdx [hunk 7]
```diff
@@ -251,6 +251,122 @@ The HTTP API implements a process for limiting and preventing excessive requests
...
+| mipsel |             |
+| ppc    |             |
+| ppc64  |             |
+| s390   |             |
+| s390x  |             |
+| x64    |             |
+
+###### Search Engine
+
+| Value      | Description              |
```
Comment (splatterxl) [inline pages/reference.mdx:363]: We should find a better way to show these self-explanatory enums

**Comment by dolfies** [comment]
Completely superseded, sorry :(

**Pull request closed without merging**